docs/man/scaladoc.1
.\" ##########################################################################
.\" #                      __                                                #
.\" #      ________ ___   / /  ___     Scala 2 On-line Manual Pages          #
.\" #     / __/ __// _ | / /  / _ |    (c) 2002-2007, LAMP/EPFL              #
.\" #   __\ \/ /__/ __ |/ /__/ __ |                                          #
.\" #  /____/\___/_/ |_/____/_/ | |    http://scala-lang.org/                #
.\" #                           |/                                           #
.\" ##########################################################################
.\"
.\" Process this file with nroff -man scala.1
.\"
.TH scaladoc 1  "May 1, 2007" "version 0.4" "USER COMMANDS"
.\"
.\" ############################## NAME ###############################
.\"
.SH NAME
.PP
\fBscaladoc\fR \- Documentation generator for the Scala 2 language
.\"
.\" ############################## SYNOPSIS ###############################
.\"
.SH SYNOPSIS
.PP
\fBscaladoc\fR  [ <\fIoptions\fR> ] <\fIsource files\fR>
.\"
.\" ############################## PARAMETERS ###############################
.\"
.SH PARAMETERS
.PP
.TP
<\fIoptions\fR>
Command line options. See \fBOPTIONS\fR below.
.TP
<\fIsource files\fR>
One or more source files to be compiled (such as MyClass.scala).
.\"
.\" ############################## DESCRIPTION ###############################
.\"
.SH DESCRIPTION
.PP
The \fBscaladoc\fR tool reads class and object definitions, written in the Scala 2 programming language, and generates their API as HTML files.
.PP
By default, the generator puts each HTML file in the same directory as its source file. You can specify a separate destination directory with \fB\-d\fR (see \fBOPTIONS\fR, below).
.PP
Supported tag comments are:
.IP
@author
.IP
@deprecated
.IP
@exception (two arguments)
.IP
@param (two arguments)
.IP
@pre
.IP
@return
.IP
@see
.IP
@since
.IP
@throws (two arguments)
.IP
@todo
.IP
@version
.PP
See also online document "How to Write Doc Comments for the Javadoc Tool" from Sun.
.\"
.\" ############################## OPTIONS ###############################
.\"
.SH OPTIONS
.PP
The generator has a set of standard options that are supported on the current development environment and will be supported in future releases.
.\"
.\" ############################## Standard Options ###############################
.\"
.SS "Standard Options"
.PP
.TP
\fB\-d\fR <\fIdirectory\fR> 
Specify where to place generated class files.
.TP
\fB\-access:<access>\fR 
Show only public, protected/public (default) or all classes and members (public,protected,private)
.TP
\fB\-windowtitle\fR <\fIwindowtitle\fR> 
Specify window title of generated HTML documentation
.TP
\fB\-doctitle\fR <\fIdoctitle\fR> 
Include title for the overview page
.TP
\fB\-stylesheetfile\fR <\fIstylesheetfile\fR> 
File to change style of the generated documentation
.TP
\fB\-header\fR <\fIpageheader\fR> 
Include header text for each page
.TP
\fB\-footer\fR <\fIpagefooter\fR> 
Include footer text for each page
.TP
\fB\-top\fR <\fIpagetop\fR> 
Include top text for each page
.TP
\fB\-bottom\fR <\fIpagebottom\fR> 
Include bottom text for each page
.TP
\fB\-version\fR 
Print product version and exit.
.TP
\fB\-help\fR 
Print a synopsis of standard options.
.\"
.\" ############################## EXAMPLES ###############################
.\"
.SH EXAMPLES
.PP
.TP
Generate documentation for a Scala program
\fBscaladoc\fR HelloWorld.scala
.TP
Generation documentation for a Scala program to the destination directory \fBclasses\fR
\fBscaladoc\fR \fB\-d\fR api HelloWorld.scala
.TP
Generate documentation for all Scala files found in the source directory \fBsrc\fR to the destination directory \fBapi\fR
\fBscaladoc\fR \fB\-d\fR api src/*.scala
.\"
.\" ############################## EXIT STATUS ###############################
.\"
.SH "EXIT STATUS"
.PP
\fBscaladoc\fR returns a zero exist status if it succeeds to process the specified input files. Non zero is returned in case of failure.
.\"
.\" ############################## AUTHOR ###############################
.\"
.SH AUTHOR
.PP
Written by Martin Odersky and other members of the Scala team.
.\"
.\" ############################## REPORTING BUGS ###############################
.\"
.SH "REPORTING BUGS"
.PP
Report bugs to <scala@listes.epfl.ch>.
.\"
.\" ############################## COPYRIGHT ###############################
.\"
.SH COPYRIGHT
.PP
This is open-source software, available to you under a BSD-like license. See accomponying "copyright" or "LICENSE" file for copying conditions. There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
.\"
.\" ############################## SEE ALSO ###############################
.\"
.SH "SEE ALSO"
.PP
\fBfsc\fR(1), \fBsbaz\fR(1), \fBscala\fR(1), \fBscalac\fR(1), \fBscalap\fR(1)


debian/changelog
scala (2.6.1-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Tue, 15 Jan 2008 18:08:31 +0100

scala (2.6.0-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Fri, 14 Sep 2007 15:22:05 +0100

scala (2.5.0-1) unstable; urgency=low

  * New upstream release
  * Depend only on java-gcj-compat, not java-gcj-compat-dev

 -- Lex Spoon <lex@debian.org>  Tue, 22 May 2007 16:51:47 +0200

scala (2.4.0-1) unstable; urgency=low

  * New upstream version.

  * Deleting all mention of the Sun JVM, just because some
    DD's find it bothersome.

  * Deleted mention of JDK5-specific classes, while awaiting
    the Sun JVM to become open source.  (Closes: #409785)

  * Added java-gcj-compat-dev as an install dependency.  (Closes: #408353)

 -- Lex Spoon <lex@debian.org>  Sun, 25 Mar 2007 11:56:18 +0200

scala (2.3.2-1) unstable; urgency=low

  * New upstream version

 -- Lex Spoon <lex@debian.org>  Fri, 19 Jan 2007 14:45:51 +0100

scala (2.3.0-1) unstable; urgency=low

  * New upstream version
  * Added a Java compiler as a build-dependency

 -- Lex Spoon <lex@debian.org>  Fri, 24 Nov 2006 16:02:09 +0100

scala (2.2.0-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Mon, 30 Oct 2006 21:35:10 -0500

scala (2.1.5-2) unstable; urgency=low

  * Rearranged to have three packages instead of four.

 -- Lex Spoon <lex@debian.org>  Wed, 28 Jun 2006 17:16:02 +0200

scala (2.1.5-1) unstable; urgency=low

  * upstream update.

 -- Lex Spoon <lex@debian.org>  Sun, 9 Jun 2006 14:45:26 +0100

scala (2.1.1-1) unstable; urgency=low

  * Initial Release.

 -- Lex Spoon <lex@debian.org>  Sun, 26 Mar 2006 14:45:26 +0100



src/manual/scala/man1/scaladoc.scala
/* NSC -- new Scala compiler
 * Copyright 2005-2007 LAMP/EPFL
 * @author Stephane Micheloud
 */
//$Id$

package scala.man1

object scaladoc extends Command {
  import _root_.scala.tools.docutil.ManPage._

  protected def cn = new Error().getStackTrace()(0).getClassName()

  val scalaLink = Link("Scala 2", "http://scala-lang.org/")

  val name = Section("NAME",

    MBold(command) & " " & NDash & " Documentation generator for the " &
    scalaLink & " language")

  val synopsis = Section("SYNOPSIS",

    CmdLine(" [ " & Argument("options") & " ] " & Argument("source files")))

  val parameters = Section("PARAMETERS",

    DefinitionList(
      Definition(
        Mono(Argument("options")),
        "Command line options. See " & Link(Bold("OPTIONS"), "#options") &
        " below."),
      Definition(
        Mono(Argument("source files")),
        "One or more source files to be compiled (such as " &
        Mono("MyClass.scala") & ").")))

  val description = Section("DESCRIPTION",

    "The " & MBold(command) & " tool reads class and object definitions, " &
    "written in the " & scalaLink & " programming language, and generates " &
    "their API as HTML files.",

    "By default, the generator puts each HTML file in the same directory as " &
    "its source file. You can specify a separate destination directory with " &
    CmdOption("d") & "(see " & Link(Bold("OPTIONS"), "#options") & ", below).",

    // tags are defined in class "scala.tools.nsc.doc.DocGenerator"
    "Supported tag comments are:",

    BulletList(
      Mono("@author"), Mono("@deprecated"),
      Mono("@exception") & " (two arguments)",
      Mono("@param") & " (two arguments)", Mono("@pre"),
      Mono("@return"), Mono("@see"), Mono("@since"),
      Mono("@throws") & " (two arguments)",
      Mono("@todo"), Mono("@version")),

    "See also online document \"" & Link("How to Write Doc Comments for the Javadoc Tool",
    "http://java.sun.com/j2se/javadoc/writingdoccomments/") & "\" from Sun.")

  val options = Section("OPTIONS",

    "The generator has a set of standard options that are supported on the " &
    "current development environment and will be supported in future releases.",

    Section("Standard Options",
      DefinitionList(
        Definition(
          CmdOption("d", Argument("directory")),
          "Specify where to place generated class files."),
        Definition(
          CmdOption("access:<access>"),
          "Show only public, protected/public (default) or all classes " &
          "and members (" & Mono("public") & ",protected,private)"),
        Definition(
          CmdOption("windowtitle", Argument("windowtitle")),
          "Specify window title of generated HTML documentation"),
        Definition(
          CmdOption("doctitle", Argument("doctitle")),
          "Include title for the overview page"),
        Definition(
          CmdOption("stylesheetfile", Argument("stylesheetfile")),
          "File to change style of the generated documentation"),
        Definition(
          CmdOption("header", Argument("pageheader")),
          "Include header text for each page"),
        Definition(
          CmdOption("footer", Argument("pagefooter")),
          "Include footer text for each page"),
        Definition(
          CmdOption("top", Argument("pagetop")),
          "Include top text for each page"),
        Definition(
          CmdOption("bottom", Argument("pagebottom")),
          "Include bottom text for each page"),
        Definition(
          CmdOption("version"),
          "Print product version and exit."),
        Definition(
          /*CmdOption("?") & "| " &*/ CmdOption("help"),
          "Print a synopsis of standard options."))))

  val examples = Section("EXAMPLES",

    DefinitionList(
      Definition(
        "Generate documentation for a Scala program",
        CmdLine("HelloWorld.scala")),
      Definition(
        "Generation documentation for a Scala program to the destination " &
        "directory " & Bold("classes"),
        CmdLine(CmdOption("d", "api") & "HelloWorld.scala")),
      Definition(
        "Generate documentation for all Scala files found in the source " &
        "directory " & Bold("src") & " to the destination directory " &
        Bold("api"),
        CmdLine(CmdOption("d", "api") & "src/*.scala"))))

  val exitStatus = Section("EXIT STATUS",

    MBold(command) & " returns a zero exist status if it succeeds to process " &
    "the specified input files. Non zero is returned in case of failure.")

  val seeAlso = Section("SEE ALSO",

    Link(Bold("fsc") & "(1)", "fsc.html") & ", " &
    Link(Bold("sbaz") & "(1)", "sbaz.html") & ", " &
    Link(Bold("scala") & "(1)", "scala.html") & ", " &
    Link(Bold("scalac") & "(1)", "scalac.html") & ", " &
    Link(Bold("scalap") & "(1)", "scalap.html"))

  def manpage = new Document {
    title = command
    date = "May 1, 2007"
    author = "Stephane Micheloud"
    version = "0.4"
    sections = List(
      name,
      synopsis,
      parameters,
      description,
      options,
      examples,
      exitStatus,
      authors,
      bugs,
      copyright,
      seeAlso)
  }
}
